// Copyright 2022 The SiliFuzz Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package silifuzz.proto;

import "proto/player_result.proto";

// A proto to store snapshot execution result identified by a snapshot ID
// and a play result.
// NextID: 5
message SnapshotExecutionResult {
  // ID of the snapshot.
  optional string snapshot_id = 1;  // semantically required.

  // Machine hostname where the snapshot was executed. Should not contain
  // the domain name part (e.g. just abc123)
  optional string hostname = 4;  // semantically required

  // Execution result from Player.
  optional PlayerResult player_result = 2;

  reserved 3;
}
